docsrc/imap/reference/manpages/systemcommands/cyr_expire.rst

```diff
@@ -42,9 +42,18 @@ There are various annotations that **cyr_expire** respects:
   messages
 - ``/vendor/cmu/cyrus-imapd/delete`` which controls the deletion of
   messages
-
-These mailbox annotations specify the age(in days) of messages in the
-given mailbox that should be expired/archived/deleted.
+- ``/vendor/cmu/cyrus-imapd/noexpire_until`` which disables the expire
+  and delete operations per user
+
+The first three mailbox annotations specify the age of messages in the
+given mailbox that should be expired/archived/deleted. The
+age is specified as a duration, the default unit are days.
```

Comment:
Need to describe the valid annotation durations here (or refer them to imapd.conf(5), which already describes them), and also need to be explicit that these are _different from the durations supported in the command line arguments_ (which are described under the heading for `-D, --delete-duration`).  

The command line durations are parsed by `parse_duration()`, a static function in cyr_expire.c, which allows and understands decimal places (e.g. `1.5d`) for whatever reason, though it reduces to integer seconds in the end.  `config_parseduration()` doesn't accept decimals because we didn't think it necessary -- the same thing can be expressed as `36h` or `1d12h`, for example (the latter of which `parse_duration()` can't handle).  So as presently implemented, valid command line durations and valid annotation/config durations are superficially similar, but aren't strictly interchangeable.

An approach here might be to just get rid of `parse_duration()`, and switch to using `config_parseduration()` for the command-line options too.  We'd need to document the change loudly when we release 3.10, in case anyone had been relying on the decimal parsing, but at least then all our "durations" will be consistent (unless there's others hiding somewhere...).  If we decide decimals are important enough to keep after all, we could probably extend `config_parseduration()` to support them.  (Though that could get messy... what should `1.5h12m` do?)

The description in imapd.conf(5) comes from here:  https://github.com/cyrusimap/cyrus-imapd/blob/master/lib/imapoptions#L87-L94.  But you can't just copy and paste that into an RST document, it's in the wrong format (for silly historical reasons).  If you want to copy and paste this text, first `make man` so that `docsrc/imap/reference/manpages/configs/imapd.conf.rst` gets generated, then copy and paste from that.

The RST syntax for linking to one of our man pages is:

```
:cyrusman:`imapd.conf(5)`
```

Reply:
Argh, thanks for this. I was totally not aware of this.

Reply:
cyr_expire is a _mess_ and it feels everything we try to clean up in it shakes out more chaos 😅

Reply:
I now replaced custom cyr_expire durations with config_parseduration: https://github.com/cyrusimap/cyrus-imapd/pull/4533/commits/cf1043174822d21a130ec9fbde9dd6c679c20e11